BUILDING.md

```diff
@@ -0,0 +1,128 @@
+Building Elasticsearch with Gradle
+=============================
+
+Elasticsearch is built using the [Gradle](https://gradle.org/) open source build tools.
+
+This document provides a general guidelines for using and working on the elasticsearch build logic.
+
+## Build logic organisation
+
+The elasticsearch project contains 3 build related projects that are included into the elasticsearch build as a [composite build](https://docs.gradle.org/current/userguide/composite_builds.html).
```

Comment:
```suggestion
The Elasticsearch project contains 3 build-related projects that are included into the Elasticsearch build as a [composite build](https://docs.gradle.org/current/userguide/composite_builds.html).
```

Reply:
In general I think it's good to capitalise "Elasticsearch" and "Elastic" in documents like this.